src/main/java/io/github/minecraftcursedlegacy/impl/data/DataStorage.java

```diff
@@ -0,0 +1,13 @@
+package io.github.minecraftcursedlegacy.impl.data;
+
+import java.util.function.Supplier;
+
+import io.github.minecraftcursedlegacy.api.data.AttachedData;
+import io.github.minecraftcursedlegacy.api.registry.Id;
+import net.minecraft.util.io.CompoundTag;
+
+public interface DataStorage {
+	CompoundTag getModdedTag();
+	void putModdedData(Id id, AttachedData data);
+	AttachedData getModdedData(Id id, Supplier<AttachedData> supplier);
```

Comment:
`getModdedData` and `putModdedData` -> `getAttachedData` and `putAttachedData`

Reply:
ah, nitpicking the impl names as well as the api.

Reply:
nothing is safe from the bikeshed.